### PR TITLE
Add note that pinning cases section is historical information.

### LINKED ISSUE #33337: Thread pinning with PostgreSQL JDBC driver
### Discussed in https://github.com/quarkusio/quarkus/discussions/33325

<div type='discussions-op-text'>

<sup>Originally posted by **davecramer** May 12, 2023</sup>
This https://quarkus.io/guides/virtual-threads#pinning-cases should now be historical with the release of 42.6.0 of the PostgreSQL JDBC driver. Virtually all synchronized methods have been replaced with reentrant lock. See https://jdbc.postgresql.org/changelogs/2023-03-17-42.6.0-release/ for details.
It would be great if you could update your site to reflect the current state.</div>

### PR DESCRIPTION
**Pull request by rolfedh**
Fixes https://github.com/quarkusio/quarkus/issues/33337

### CODE REVIEW

File: docs/src/main/asciidoc/virtual-threads.adoc [hunk 1]
```diff
@@ -141,6 +141,8 @@ Even worse, lots of widespread libraries are pinning and would require code chan
 
 For more information, see link:https://quarkus.io/blog/virtual-thread-1/[When Quarkus meets Virtual Threads]
 
+IMPORTANT: With the release of 42.6.0 of the PostgreSQL JDBC driver, this information about pinning cases is historical . Virtually all synchronized methods have been replaced with reentrant lock. For more information, see the link:https://jdbc.postgresql.org/changelogs/2023-03-17-42.6.0-release/[Notable Changes] for PostgreSQL JDBC driver 42.6.0.
```
Comment (gsmet) [inline docs/src/main/asciidoc/virtual-threads.adoc:144]:
```suggestion
IMPORTANT: With the release of 42.6.0 of the PostgreSQL JDBC driver, this information about pinning cases is historical. Virtually all synchronized methods have been replaced with reentrant locks. For more information, see the link:https://jdbc.postgresql.org/changelogs/2023-03-17-42.6.0-release/[Notable Changes] for PostgreSQL JDBC driver 42.6.0.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I will let @cescoffier review it but these should be fixed.

**Comment by davecramer** [comment]
LGTM! Thanks

**Comment by rolfedh** [comment]
Hi, @cescoffier. Please review and approve. Thanks.

File: docs/src/main/asciidoc/virtual-threads.adoc [hunk 2]
```diff
@@ -137,10 +137,14 @@ It can be reasonably easy to avoid these situations in your code, but verifying
 Typically, while experimenting with virtual threads, we realized that old versions of the link:{pgsql-driver}[postgresql-JDBC driver]
 results in frequent pinning.
 Most JDBC drivers still pin the carrier thread.
-Even worse, lots of widespread libraries are pinning and would require code changes.
+Even worse, many libraries require code changes.
 
 For more information, see link:https://quarkus.io/blog/virtual-thread-1/[When Quarkus meets Virtual Threads]
 
+IMPORTANT: This information about pinning cases applies to PostgreSQL JDBC driver 42.5.4 and earlier.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/virtual-threads.adoc:144]: I think the position is misleading. I read it like: Oh! Cool, they fix all the pining.
It's not the case. I would just replace `old versions` with `versions preceding 42.6.0`

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/virtual-threads.adoc (hunk 2 above)
Comment (rolfedh) [reply docs/src/main/asciidoc/virtual-threads.adoc:143]: Thanks, @cescoffier. Is this the change you were suggesting?

Comment (cescoffier) [reply docs/src/main/asciidoc/virtual-threads.adoc:143]: > Typically, while experimenting with virtual threads, we realized that old versions of the link:{pgsql-driver}[postgresql-JDBC driver]

would become:

> Typically, while experimenting with virtual threads, we realized that versions older than 42.6.0 of the link:{pgsql-driver}[postgresql-JDBC driver]

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Review by cescoffier — APPROVED** [review]

**Comment by rolfedh** [comment]
@gsmet, would you please merge this PR? Thanks.

**Pull request merged**